src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.T.cs

```diff
@@ -488,6 +490,249 @@ public static void Ceiling<T>(ReadOnlySpan<T> x, Span<T> destination)
+                ((IsUInt64Like<TFrom>() || IsInt64Like<TFrom>()) && (IsUInt64Like<TTo>() || IsInt64Like<TTo>())))
+            {
+                source.CopyTo(Rename<TTo, TFrom>(destination));
+                return;
+            }
+
+            if (typeof(TFrom) == typeof(float) && IsUInt32Like<TTo>())
+            {
+                InvokeSpanIntoSpan<float, uint, ConvertSingleToUInt32>(Rename<TFrom, float>(source), Rename<TTo, uint>(destination));
+                return;
+            }
+
+            if (typeof(TFrom) == typeof(float) && IsInt32Like<TTo>())
+            {
+                InvokeSpanIntoSpan<float, int, ConvertSingleToInt32>(Rename<TFrom, float>(source), Rename<TTo, int>(destination));
+                return;
+            }
+
+            if (typeof(TFrom) == typeof(double) && IsUInt64Like<TTo>())
+            {
+                InvokeSpanIntoSpan<double, ulong, ConvertDoubleToUInt64>(Rename<TFrom, double>(source), Rename<TTo, ulong>(destination));
+                return;
+            }
+
+            if (typeof(TFrom) == typeof(double) && IsInt64Like<TTo>())
+            {
+                InvokeSpanIntoSpan<double, long, ConvertDoubleToInt64>(Rename<TFrom, double>(source), Rename<TTo, long>(destination));
+                return;
+            }
+
+            if (typeof(TFrom) == typeof(ushort) && typeof(TTo) == typeof(byte))
+            {
+                InvokeSpanIntoSpan_2to1<ushort, byte, NarrowUInt16ToByteOperator>(Rename<TFrom, ushort>(source), Rename<TTo, byte>(destination));
+                return;
+            }
+
+            if (typeof(TFrom) == typeof(short) && typeof(TTo) == typeof(sbyte))
+            {
+                InvokeSpanIntoSpan_2to1<short, sbyte, NarrowInt16ToSByteOperator>(Rename<TFrom, short>(source), Rename<TTo, sbyte>(destination));
+                return;
+            }
+
+            if (IsUInt32Like<TFrom>() && typeof(TTo) == typeof(ushort))
+            {
+                InvokeSpanIntoSpan_2to1<uint, ushort, NarrowUInt32ToUInt16Operator>(Rename<TFrom, uint>(source), Rename<TTo, ushort>(destination));
+                return;
+            }
+
+            if (IsInt32Like<TFrom>() && typeof(TTo) == typeof(short))
+            {
+                InvokeSpanIntoSpan_2to1<int, short, NarrowInt32ToInt16Operator>(Rename<TFrom, int>(source), Rename<TTo, short>(destination));
+                return;
+            }
+
+            if (IsUInt64Like<TFrom>() && IsUInt32Like<TTo>())
+            {
+                InvokeSpanIntoSpan_2to1<ulong, uint, NarrowUInt64ToUInt32Operator>(Rename<TFrom, ulong>(source), Rename<TTo, uint>(destination));
+                return;
+            }
+
+            if (IsInt64Like<TFrom>() && IsInt32Like<TTo>())
+            {
+                InvokeSpanIntoSpan_2to1<long, int, NarrowInt64ToInt32Operator>(Rename<TFrom, long>(source), Rename<TTo, int>(destination));
+                return;
+            }
```

Comment:
I'm sure there are more conversions we could vectorize here, but this list seemed to be a good starting place.  Hopefully there's a better way to express groups of them, but nothing was apparent to me.